### ISSUE TITLE
[manual] add gnuplot files `parametric-example-cut` to repository?

### ISSUE THREAD

**Issue by Mo-Gul** [pgf-tikz/pgf#719]
Shall the gnuplot files
- pgf-parametric-example-cut.gnuplot
- pgf-parametric-example-cut.table

in the folder <doc\generic\pgf\version-for-luatex\en\plots> from
https://github.com/pgf-tikz/pgf/blob/f5d0ffd6e4a553d40855370c692fea6e33a0026d/doc/generic/pgf/text-en/pgfmanual-en-tikz-plots.tex#L370-L378
be added to the repository as well?

**Comment by hmenke** [comment]
Are there any problems if those files are missing?  I though the data files were only needed if the plot used the external `gnuplot` program.

**Comment by Mo-Gul** [comment]
These files are *created* by `gnuplot`. And `gnuplot` *is* used in the above example to create the plot. (For details see sections 22.2 and 22.6 in the pgfmanual.)

To answer your question: No, there is no problem if you have `gnuplot` installed and working. But if not, then the `.table` file can used instead using `\tikz \draw plot [...] file {<file name>.table};`.

In this case I see two options to be *consistent* (again): Either add the above files, too, or delete all the others.

Another pro point is that these files are some kind of a test. Thus the `.gnuplot` files are tests that TikZ/PGF still does what it should and the `.table` files are tests that `gnuplot` still does what it should.

**Comment by hmenke** [comment]
Okay, I will add them then.  I also noticed that some gnuplot files are overwritten when building the documentation.  This might have to do with the gnuplot version change from 4.x to 5.x some time in the last years.

**Comment by Mo-Gul** [comment]
Thanks.

The gnuplot files should be overwritten each time you compile, i.e. when you compile with LuaLaTeX the files in `doc/generic/pgf/version-for-luatex/en/plots` will *all* be recreated. Of course the same holds for the other engines.

Thus, I don't know if/when the gnuplot files under `doc/generic/pgf/text-en/plots` are created/used. Maybe this is a leftover when it was allowed to right in higher folders than the main file? This would be a perfect explanation, because all the plot files in all the different folders are identical ...

**Comment by muzimuzhi** [comment]
Commit https://github.com/pgf-tikz/pgf/commit/f7a24c56adf7de3c53ba9707d35650b542ee453a only added file `pgf-parametric-example-cut.table` under path for driver `dvipdfmx`. See https://github.com/pgf-tikz/pgf/blob/master/doc/generic/pgf/version-for-dvipdfmx/en/plots/pgf-parametric-example-cut.table. For other drivers, especially `luatex`, this file is still missing.

Therefore `texlua build.lua manual luatex` will write following warning to log,
```
Package pgf Warning: Plot data file `plots/pgf-parametric-example-cut.table' not found. on input line 1.
```
and cause the corresponding drawing missing from the output PDF (`pgfmanual` v3.1.6a, page 348).
![image](https://user-images.githubusercontent.com/6376638/95663409-a6a56800-0b71-11eb-8f3d-2f1a8fec2ce6.png)

This can be fixed either by copying/linking that file to all drivers, or installing `gnuplot` and enabling shell-escape in CI.

**Comment by Mo-Gul** [comment]
Unfortunately a fix to this issue didn't make it into v3.1.7 ...

@muzimuzhi, do you have a clue how to extend the build script to also "install" gnuplot so the `.gnuplot` and `.table` files are generated on the fly? (This should be the long-term solution so also changes in gnuplot will be detected when the test-suite is working.)

If also Henri doesn't have a clue how to do this or doesn't find the time to implement it "soon" the temporary fix should be to (re)add the `.table` file(s).

**Comment by muzimuzhi** [comment]
Fixed in https://github.com/pgf-tikz/pgf/commit/eee4a6a9bce510d24eee227af7bce3b0ffc620d0 which is included from 3.1.7a, so I will close this.
